### PR TITLE
[Core] Updates to multipart support

### PR DESCRIPTION
**Pull request by annatisch**
Includes fixes for #10484 and #10485
Also contributes to #9534

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

**Comment by annatisch** [comment]
FYI @pakrym

File: sdk/core/azure-core/azure/core/pipeline/_base.py [hunk 1]
```diff
@@ -194,7 +194,8 @@ def run(self, request, **kwargs):
         :return: The PipelineResponse object
         :rtype: ~azure.core.pipeline.PipelineResponse
         """
-        self._prepare_multipart_mixed_request(request)
+        multipart_options = kwargs.pop("multipart_options", None) or {}
```
Comment (lmazuel) [inline sdk/core/azure-core/azure/core/pipeline/_base.py:197]: Two things:
- Why not `kwargs.pop("multipart_options", {})` ?
- Do we want this parameter to be hidden?

**Comment by annatisch** [comment]
Separated first part out into separate PR to be merged first:
https://github.com/Azure/azure-sdk-for-python/pull/10616

**Comment by Petermarcu** [comment]
Is this one done or still needed? I saw it referenced in another bug that sounded completed.

**Comment by annatisch** [comment]
Thanks @Petermarcu - this is still needed (though part of it was moved out into another PR and merged). I will get this one updated and ready for review.

**Pull request closed without merging**